lib/segment/src/segment/snapshot.rs

```diff
@@ -422,11 +422,16 @@ fn updated_files(old: &SegmentManifest, current: &SegmentManifest) -> HashSet<Pa
         //    - if file is versioned in *one* of the manifests only, compare *file* version against
         //      other *segment* version
         //    - if file is versioned in both manifests, compare file versions
-        let is_updated = old_version.or_segment_version(old.segment_version)
-            < current_version.or_segment_version(current.segment_version);
+        if old_version < current_version {
+            updated.insert(path.to_path_buf());
+            continue;
+        }
 
-        if is_updated {
-            updated.insert(path.clone());
+        // 3. if `old` manifest contains this file and file/segment versions in both `old` and `current` manifests are 0
+        //
+        // TODO: Add explanation 🙄
```

Comment:
Yeah

Reply:
My bad. This works around the issue when we can't distinguish between new empty segment (version 0) and segment where we applied the very first operation in the WAL (also version 0). If some file is at version 0 in both manifests, we always include it into snapshot (so that we can guarantee that we always included changes from operation 0).

I'll add comment tomorrow.

Reply:
Added.